src/main/scala/io/github/kiemlicz/shelm/ChartDownloader.scala

```diff
@@ -40,13 +97,28 @@ object ChartDownloader {
         downloadDir / name
       case ChartLocation.RemoteRepository(ChartName(name), uri, settings, chartVersion) =>
         val authOpts = HelmPlugin.chartRepositoryCommandFlags(settings)
-        val allOptions = s"--repo $uri $name $authOpts -d $downloadDir${chartVersion.map(v => s" --version $v").getOrElse("")} --untar"
+        val allOptions = s"--repo $uri $name $authOpts -d $downloadDir${
+          chartVersion.map(v => s" --version $v").getOrElse("")
+        } --untar"
         IO.delete(downloadDir)
         pullChart(allOptions, sbtLogger)
         downloadDir / name
     }
   }
 
+  def cleanCache(cacheBaseDir: File, noVersionToKeep: Int, logger: Logger): Int = {
+//    val numOfCleanedDirs = cacheBaseDir.listFiles()
+//      .flatMap(f => f.listFiles())
+//      .flatMap(f => f.listFiles().sortBy(g => g.attributes.lastModifiedTime()).dropRight(noVersionToKeep)).map(f => {
+//      logger.info(s"Removing helm cache directory: $f")
+//      IO.delete(f)
+//    }
+//    ).length
+//    cacheBaseDir.listFiles().filter(f => f.listFiles().length == 0).foreach(_.delete())
+//    numOfCleanedDirs
+    0
+  }
+
```

Comment:
I don't know what should be semantics of cache cleaning now  
Should it be just deleting whole cache directory? or some smart "the earliest downloaded" calculation?

Reply:
I think we can start with your current version: `cacheBaseDir.listFiles().foreach(IO.delete)`